tests/containers/kubectl.pm

```
@@ -0,0 +1,136 @@
+
+    # Configure CSP/k3s. Only one CSP at a time is possible due to the conflicting configuration in .kube/config
+    assert_script_run('mkdir -p ~/.kube');
+    my $provider = get_var("KUBECTL_CLUSTER", "k3s");
+    if ($provider eq "k3s") {
+        install_k3s();
+
+        # Ensure k3s has not installed its own kubectl
+        assert_script_run("! stat /usr/local/bin/kubectl");
+        validate_script_output("whereis kubectl", sub { $_ !~ m/\/usr\/local\/bin/ });
+    } else {
+        die "Invalid KUBECTL_CLUSTER defined";
+    }
+
+    # Function to check if the output either has no resources or has at least two lines
+    # Two lines because one line is the header and the following lines are the data rows
+    validate_script_output("kubectl cluster-info", qr/Kubernetes control plane/);
+    record_info('Test get', 'State tests (get commands)');
+    # Check for default namespaces to be present
+    validate_script_output("kubectl get namespaces", qr/default/);
+    validate_script_output("kubectl get namespaces", qr/system/);
+    # Check service output
+    validate_script_output("kubectl get service --all-namespaces", sub { $_ =~ m/No resources found/ || split(/\n/, $_) > 1; });
+    # At least the coredns pod must be present
+    validate_script_output("kubectl get pods --all-namespaces", sub { $_ =~ m/coredns/ });
+    # There must be at least one endpoint present
+    validate_script_output("kubectl get endpoints", sub { split(/\n/, $_) > 1; });
+    # There must be at least the coredns deployment present
+    validate_script_output("kubectl get deployments --all-namespaces", sub { $_ =~ m/coredns/ });
+    # Check if more than 1 replicasets are present
+    validate_script_output("kubectl get replicasets --all-namespaces --no-headers", sub { split(/\n/, $_) > 1; });
+    # Check ingresses, those are empty by default
+    validate_script_output("kubectl get ingresses --all-namespaces --no-headers", sub { $_ =~ m/No resources found/ });
```

Again, what do we gain by checking there is no ingress?

---

Same as above - perliminary smoke test to allow to fail fast.